src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,32 @@
+package seedu.addressbook.data.person;
+
+/** 
+ * Store information for a contact
+ */
+
+public abstract class Contact {
+	
+    public String value;
+    protected boolean isPrivate;
+	
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
This is fine, but note that now an Email and Address having the same value will be considered equal even if they are of different types.

You can override it at child classes but make sure to override the hashCode() as well.